### ISSUE TITLE
docker_container should support HostConfig/Init

### ISSUE THREAD

**Issue by rmoriz** [chef-cookbooks/docker#934]
Docker v13.x+ implemented "tini" https://github.com/krallin/tini#using-tini and removes a lot of redundancy in images itself.

```shell
# without
➜  docker run --rm --name foo  -it debian:jessie 
root@189c48c24781:/# ps auxww
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.1  20244  3028 pts/0    Ss   12:10   0:00 bash
root         7  0.0  0.1  17500  2136 pts/0    R+   12:14   0:00 ps auxww

# with --init
➜  docker run --rm --name foo --init -it debian:jessie
root@296166a5c4bc:/# ps auxww
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  1.0  0.0   1052     4 pts/0    Ss   12:14   0:00 /dev/init -- bash
root         6  0.0  0.1  20244  3048 pts/0    S    12:14   0:00 bash
root         7  0.0  0.1  17500  2080 pts/0    R+   12:14   0:00 ps auxww
```

It's poorly documented, the API doc misses it but it's implemented using 
```json
"HostConfig": {
…
"Init": true
…
}
```

**Comment by chasebolt** [comment]
released in `2.16.3`

**Comment by rmoriz** [comment]
thanks 👍
